wasp-app-runner/src/waspCli.ts

```diff
@@ -0,0 +1,121 @@
+import { stripVTControlCharacters } from "node:util";
+
+import { log } from "./logging.js";
+import { getWaspcDirAbsPath } from "./path.js";
+import { processManager } from "./process.js";
+import { DbType } from "./db/index.js";
+
+export function migrateDb({
+  waspCliCmd,
+  pathToApp,
+  extraEnv,
+}: {
+  waspCliCmd: string;
+  pathToApp: string;
+  extraEnv: Record<string, string>;
+}): Promise<number | null> {
+  return processManager.spawnWithLog({
+    name: "migrate-db",
+    cmd: waspCliCmd,
+    args: ["db", "migrate-dev"],
+    cwd: pathToApp,
+    extraEnv,
+  });
+}
+
+export function startApp({
+  waspCliCmd,
+  pathToApp,
+  extraEnv,
+}: {
+  waspCliCmd: string;
+  pathToApp: string;
+  extraEnv: Record<string, string>;
+}): Promise<number | null> {
+  return processManager.spawnWithLog({
+    name: "start-app",
+    cmd: waspCliCmd,
+    args: ["start"],
+    cwd: pathToApp,
+    extraEnv,
+  });
+}
+
+export async function getAppInfo({
+  waspCliCmd,
+  pathToApp,
+}: {
+  waspCliCmd: string;
+  pathToApp: string;
+}): Promise<{
+  appName: string;
+  dbType: DbType;
+}> {
+  const { stdoutData, exitCode } = await processManager.spawnAndCollectStdout({
+    name: "get-app-info",
+    cmd: waspCliCmd,
+    args: ["info"],
+    cwd: pathToApp,
+  });
+  const stdoutDataWithoutAnsiChars = stripVTControlCharacters(stdoutData);
+
+  if (exitCode !== 0) {
+    log(
+      "get-app-info",
+      "error",
+      `Failed to get app info: ${stdoutDataWithoutAnsiChars}`
+    );
+    process.exit(1);
+  }
+
+  const appNameMatch = stdoutDataWithoutAnsiChars.match(/Name: (.*)$/m);
+  const dbTypeMatch = stdoutDataWithoutAnsiChars.match(
+    /Database system: (.*)$/m
+  );
+
+  if (!appNameMatch) {
+    log("get-app-info", "error", `Failed to get app name`);
+    process.exit(1);
+  }
+
+  if (!dbTypeMatch) {
+    log("get-app-info", "error", `Failed to get database type`);
+    process.exit(1);
+  }
```

Comment:
Can we check for `null` explicitly?

I am generally not a fan of coercions in boolean expressions. I wouldn't normally mention it, but in this case, we're dealing with regexes which can match empty strings.

Of course, if this happens, something else is probably wrong. Still, it's best if our conditions communicate exactly what we want to say. And in this case, I believe we want to say "exit if you haven't found a match".